enterprise/backend/test/metabase_enterprise/serialization/v2/data_apps_test.clj

```diff
@@ -0,0 +1,148 @@
+              (is (zero? (t2/count :model/DataAppDefinition :app_id (:id imported-app))))
+              (is (=? {:creator_id (mt/user->id :crowberto)
+                       :description "A test data app without released definition"
+                       :name        "Test Data App No Release"
+                       :slug        "test-data-app-no-release"
+                       :status      :private}
+                      imported-app)))))))))
+
+(deftest data-app-multiple-definitions-one-release-test
+  (testing "DataApp with multiple definitions and one release will only serialize the relased definition"
+    (ts/with-random-dump-dir [dump-dir "serdes-data-apps-multiple-"]
+      (ts/with-dbs [source-db dest-db]
+        (ts/with-db source-db
+          ;; Create data app with initial definition
+          (data-apps.tu/with-data-app!
+            [app {:name       "Test Data App Multiple"
+                  :slug       "test-data-app-multiple"
+                  :description "A test data app with multiple definitions"
+                  :definition {:config data-apps.tu/default-app-definition-config}}]
+            ;; Create a second definition and release it
+            (let [new-definition {:config {:actions []
+                                           :pages [{:name "Released Page"}]
+                                           :parameters []}
+                                  :creator_id (mt/user->id :crowberto)}
+                  _ (data-apps.models/set-latest-definition! (:id app) new-definition)
+                  _ (data-apps.models/release! (:id app) (mt/user->id :crowberto))
+                  ;; Create another definition after release (not released)
+                  _ (data-apps.models/set-latest-definition! (:id app)
+                                                             {:config {:actions []
+                                                                       :pages [{:name "Post-Release Draft"}]
+                                                                       :parameters []}
+                                                              :creator_id (mt/user->id :crowberto)})
+                  extraction (serdes/with-cache (into [] (extract/extract {})))
+                  data-apps  (filter #(= "DataApp" (-> % :serdes/meta last :model)) extraction)
+                  data-app   (first data-apps)]
+              (is (=? {:creator_id          "crowberto@metabase.com"
+                       :name                "Test Data App Multiple"
+                       :released_definition [{:config          (mt/malli=? :map)
+                                              :creator_id      "crowberto@metabase.com"
+                                              :release         [{:app_id     (:entity_id data-app)
+                                                                 :creator_id "crowberto@metabase.com"
+                                                                 :retracted false}]
```

Comment:
The ordering `app -> definition -> release` feels clumsy to me, and it also bothers me that we repeat `app_id` inside `release`.

I was imagining `app -> release -> definition`:

```clojure
{:name "My App"
 :creator_id "my@email.com"
 :release {:creator_id "your@email.com"
           :definition {:creator_id "their@email.com"
                        :config map?}}}
```

I realize looking at the code that this might be fighting the opinionated magic. It's not worth swimming too hard against the current, but as it stands it feels like the snapshot implementation details are leaking out in a weird way.

Reply:
It's similar to how we think of git branches having commits, rather than commits having branches. Especially since we're not going to serialize any history for the single branch we export, it feels more natural this way.